modules/oe_webtools_cookie_consent/oe_webtools_cookie_consent.module

```diff
@@ -25,3 +27,22 @@ function oe_webtools_cookie_consent_preprocess_media_oembed_iframe(array &$varia
     }
   }
 }
+
+/**
+ * Implements hook_page_attachments().
+ */
+function oe_webtools_cookie_consent_page_attachments(array &$attachments) {
+  $event = new CookieConsentEvent();
+  $event_dispatcher = \Drupal::service('event_dispatcher');
+  $event_dispatcher->dispatch(CookieConsentEvent::NAME, $event);
```

Comment:
I'm ok with this dispatching this event so that sites can disable the CCK in certain pages if they need to, but I don't know if we need to have an event on our own module. We can check the configuration value directly here and set the default value in the event object directly before dispatching it. What do you think?

Reply:
So, I discussed with @upchuk . It's ok to do it both ways, even if we slightly prefer to load the config here.
But anyway we'll go for yet another approach 😈 
I challenged the idea of having a configuration to enable/disable the functionality, but it seems there might be cases where this is required to be changed by users (and we want to avoid to have them disable modules).
But at the moment we are only allowing to disable the popup part. Let's do it right (also as it has been done by the [NextEuropa team](https://github.com/ec-europa/platform-dev/pull/2636)) and put two configurations: one to enable the popup and one to enable the video popup.
Since we need to check if it's enabled/disabled in two functions already, we have to introduce a service to easily retrieve the setting. The service will take care of loading the config (so we get rid of the default subscriber). But we will leave the event dispatching, as we want to leave the possibility to alter the default setting (based on page, role, video service, etc).
So the service will have two methods, one to check if the page popup is enabled, and it throws an event.
Another method will check if the video popup is enabled, and it has one parameter which is the video url itself. So we can send this into another event.